AngouriMath/Experimental/Limits/Limit.cs

```diff
@@ -12,18 +12,63 @@ public enum ApproachFrom
     }
     public static class Limit
     {
-        internal static Entity? ComputeLimitToInfinity(Entity expr, VariableEntity x)
+        private static (Entity? result, bool isFinite) ComputeLimitToInfinityForwarder(Entity expr, VariableEntity x)
+        {
+            expr = expr.Simplify();
+            return ComputeLimitToInfinity(expr, x);
+        }
+        internal static (Entity? result, bool isFinite) ComputeLimitToInfinity(Entity expr, VariableEntity x)
         {
             var substitutionResult = LimitSolvers.SolveBySubstitution(expr, x);
-            if (substitutionResult != null) return substitutionResult;
+            if (substitutionResult.result is {}) return substitutionResult;
 
-            var polynomialResult = LimitSolvers.SolvePolynomialDivision(expr, x);
-            if (polynomialResult != null) return polynomialResult;
+            var logarithmResult = LimitSolvers.SolveAsLogarithm(expr, x);
+            if (logarithmResult.result is {}) return logarithmResult;
 
-            return null;
+            var polynomialResult = LimitSolvers.SolveAsPolynome(expr, x);
+            if (polynomialResult.result is {}) return polynomialResult;
+
+            var polynomialDivisionResult = LimitSolvers.SolvePolynomialDivision(expr, x);
+            if (polynomialDivisionResult.result is {}) return polynomialDivisionResult;
+            
+            var logarithmDivisionResult = LimitSolvers.SolveAsLogarithmDivision(expr, x);
+            if (logarithmDivisionResult.result is {}) return logarithmDivisionResult;
+
+            return (null, false);
         }
+
         public static Entity? ComputeLimit(Entity expr, VariableEntity x, Entity dist, ApproachFrom side)
         {
+            expr = expr.DeepCopy();
+            return _ComputeLimitSwitcher(expr, x, dist, side).result;
+        }
+
+        internal static (Entity? result, bool isFinite) _ComputeLimitSwitcher(Entity expr, VariableEntity x,
+            Entity dist, ApproachFrom side)
+        {
+            var lim = _ComputeLimit(expr, x, dist, side);
+            if (!(lim.result is null))
+                return lim;
+            // theoretically, for cases such limit (x -> -1) 1 / (x + 1)
+            // this method will return NaN, but thanks to replacement of x to an non-definite expression,
+            // it is somehow compensated
+
+            for (int i = 0; i < expr.ChildrenCount; i++)
+            {
+                var child = expr.GetChild(i);
+                var (limit, isFinite) = _ComputeLimitSwitcher(child, x, dist, side);
+                if (limit is {} && isFinite)
```

Comment:
Can condense two lines into `if (_ComputeLimitSwitcher(child, x, dist, side) is var ({ } limit, true))`

Reply:
As far as I know, @MomoDeve is planning to get rid of this return type `(Entity?, bool)`, so it will be refactored anyway